File: nysol_python/mcmd/methods/convTex/db/smp/msep.py
# ================================================
# コマンドマニュアル自動作成用サンプルコードデータ
# ================================================
db={}
db['name']='msep'

############################### IDATA
db['idatas']=[]

data={}
data['name']='dat1.csv'
data['text']='''
item,date,quantity,price
A,20081201,1,10
B,20081201,4,40
A,20081202,2,20
A,20081203,3,30
B,20081203,5,50
'''
db['idatas'].append(data)

############################### SCRIPTS
db['scripts']=[]

script={}
script['title']='基本例'
script['comment']='''
``dat`` という名前のディレクトリを作成し、
そのディレクトリに日付項目値 ``date`` ごとに異なるファイルに出力する。
'''
script['sh_code']='''
msep d='./dat/${date}.csv' -p i=dat1.csv
'''
script['py_code']='''
nm.msep(d="./dat/${date}.csv", p=True, i="dat1.csv").run()
'''
script['odatas']=''
db['scripts'].append(script)

File: olddoc/mcmd/en/examples/msep.rb
#!/usr/bin/env ruby
# coding: utf-8
require "./mkTex.rb"

File.open("dat1.csv","w"){|fpw| fpw.write(
<<'EOF'
item,date,quantity,price
A,20081201,1,10
B,20081201,4,40
A,20081202,2,20
A,20081203,3,30
B,20081203,5,50
EOF
)}

############## Example 1
title="Basic Example"
comment=<<'EOF'
Create a directory named \verb|dat|, and separate the data according to the \verb|date| value in the directory.
EOF
scp=<<'EOF'
more dat1.csv
msep d='./dat/${date}.csv' -p i=dat1.csv
ls ./dat
more ./dat/20081201.csv
more ./dat/20081202.csv
more ./dat/20081203.csv
more ./dat/B.csv
EOF
system("rm -rf dat")
run(scp,title,comment)

############## Example 2
#title="Error if data is not sorted"
#comment=<<'EOF'
#The command will not work properly if \verb|item| field is not sorted beforehand.
#EOF
#scp=<<'EOF'
#msep d='./dat/${item}.csv' -p i=dat1.csv
#ls ./dat
#more ./dat/A.csv
#more ./dat/B.csv
#EOF
#system("rm -rf dat")
#run(scp,title,comment)
